lib/mix/tasks/phx.gen.auth.ex

```diff
@@ -268,35 +273,146 @@ defmodule Mix.Tasks.Phx.Gen.Auth do
     end
   end
 
-  defp prompt_for_conflicts(context) do
-    context
+  defp scope_config(context, requested_scope) do
+    existing_scopes = Mix.Phoenix.Scope.scopes_from_config(context.context_app)
+
+    {_, default_scope} =
+      Enum.find(existing_scopes, {nil, nil}, fn {_, scope} -> scope.default end)
+
+    key = String.to_atom(requested_scope || context.schema.singular)
+
+    {create_new?, scope, config_string} =
+      if Map.has_key?(existing_scopes, key) do
+        {false, existing_scopes[key], nil}
+      else
+        {true, new_scope(context, key, default_scope),
+         scope_config_string(context, key, default_scope)}
+      end
+
+    %{
+      scopes: existing_scopes,
+      default_scope: default_scope,
+      create_new?: create_new?,
+      scope: scope,
+      config_string: config_string
+    }
+  end
+
+  defp new_scope(context, key, default_scope) do
+    Mix.Phoenix.Scope.new!(key, %{
+      default: !default_scope,
+      module: Module.concat([context.module, "AuthScope"]),
+      fixture:
+        {Module.concat([context.module, "Fixtures"]),
+         :"register_and_log_in_#{context.schema.singular}"},
+      assign_key: :current_scope,
+      access_path: [
+        String.to_atom(context.schema.singular),
+        context.schema.opts[:primary_key] || :id
+      ],
+      schema_key:
+        String.to_atom("#{context.schema.singular}_#{context.schema.opts[:primary_key] || :id}"),
+      schema_type: if(context.schema.binary_id, do: :binary_id, else: :id),
+      schema_table: context.schema.table
+    })
+  end
+
+  defp scope_config_string(context, key, default_scope) do
+    """
+    config :#{context.context_app}, :scopes,
+      #{key}: [
+        default: #{if default_scope, do: false, else: true},
+        module: #{inspect(context.module)}.AuthScope,
+        fixture: {#{inspect(context.module)}Fixtures, :register_and_log_in_#{context.schema.singular}},
+        assign_key: :current_scope,
+        access_path: [:#{context.schema.singular}, :#{context.schema.opts[:primary_key] || :id}],
+        schema_key: :#{context.schema.singular}_#{context.schema.opts[:primary_key] || :id},
+        schema_type: :#{if(context.schema.binary_id, do: :binary_id, else: :id)},
+        schema_table: :#{context.schema.table}
+      ]\
+    """
+  end
+
+  defp prompt_for_conflicts(binding) do
+    prompt_for_scope_conflicts(binding)
+
+    binding
     |> files_to_be_generated()
     |> Mix.Phoenix.prompt_for_conflicts()
   end
 
-  defp files_to_be_generated(%Context{schema: schema, context_app: context_app} = context) do
+  defp prompt_for_scope_conflicts(binding) do
+    schema = binding[:schema]
+    %{scope: scope, default_scope: default_scope, create_new?: new?} = binding[:scope_config]
+
+    cond do
+      # TODO: check if we want to include this or only ask for the existing default case
+      scope && not new? ->
```

Comment:
I would try to keep it simpler and just pick a different name?

Reply:
I would try user, then accounts_user, then my_app_accounts_user. If all of these are taken, then give up :P

Reply:
alright!